amundsen_application/static/js/components/Feedback/FeedbackForm/index.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import LoadingSpinner from '../../common/LoadingSpinner';
+
+// TODO: Use css-modules instead of 'import'
+import './styles.scss';
+import { ResetFeedbackRequest, SubmitFeedbackRequest } from "../../../ducks/feedback/types";
+
+import { SendingState } from '../types';
+
+interface FeedbackFormState {
+  sendState: SendingState;
+}
+
+export interface StateFromProps {
+  sendState: SendingState;
+}
+
+export interface DispatchFromProps {
+  submitFeedback: (data: FormData) => SubmitFeedbackRequest;
+  resetFeedback: () => ResetFeedbackRequest;
+}
+
+type FeedbackFormProps = StateFromProps & DispatchFromProps;
+
+abstract class AbstractFeedbackForm extends React.Component<FeedbackFormProps, FeedbackFormState> {
+  public static defaultProps: FeedbackFormProps = {
+    sendState: SendingState.IDLE,
+    submitFeedback: () => undefined,
+    resetFeedback: () => undefined,
+  };
+
+  static FORM_ID = "feedback-form";
+
+  protected constructor(props) {
+    super(props);
+
+    this.state = {
+      sendState: this.props.sendState
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
You're correct. This pattern exists because of my initial misunderstanding of getDerivedStateFromProps and we will have one PR in the future to clean up all the spots where we have this redundancy.